### PR TITLE
chore: use npm 8 MONGOSH-975

### PR DESCRIPTION
**Pull request by addaleax**

### CODE REVIEW

File: .evergreen/evergreen.yml.in [hunk 1]
```diff
@@ -121,6 +121,24 @@ const EXECUTABLE_PKG_INFO = [
...
+        tar cvzf npm-logs.tgz npm-logs
+  - command: s3.put
+    params:
+      aws_key: ${aws_key}
+      aws_secret: ${aws_secret}
+      local_file: npm-logs.tgz
+      remote_file: mongosh/binaries/${revision}/${revision_order_id}/npm-logs-${build_variant}-${task_name}-${task_id}.tgz
+      bucket: mciuploads
+      permissions: public-read
+      content_type: application/x-gzip
```
Comment (addaleax) [inline .evergreen/evergreen.yml.in:140]: Uploading all npm logs from a run was helpful for debugging, I’d just keep it here.

File: packages/async-rewriter2/src/async-writer-babel.ts [hunk 2]
```diff
@@ -72,7 +72,9 @@ export default class AsyncWriter {
         ]
       ], { code: true, ast: false })?.code as string;
     } catch (e: any) {
-      e.message = e.message.replace('unknown: ', '');
+      const { message } = e;
+      delete e.message; // e.message may have been non-writable
+      e.message = message.replace('unknown: ', '');
```
Comment (addaleax) [inline packages/async-rewriter2/src/async-writer-babel.ts:77]: This was a result from one of the implicit dependency updates that came with regenerating the package-lock files

File: scripts/docker/rocky8-package.Dockerfile [hunk 3]
```diff
@@ -9,5 +9,8 @@ RUN dnf -y install python3 rpm-build dpkg-devel dpkg-dev git
 # Add Node.js
 RUN curl -sL https://rpm.nodesource.com/setup_14.x | bash -
 RUN dnf install -y nodejs
+RUN npm i -g npm@8.x
+RUN mkdir -p /tmp/home
+ENV HOME=/tmp/home
```
[the remark this answers is not in the document]
Comment (Anemy) [reply scripts/docker/rocky8-package.Dockerfile:14]: Does it make sense to  leave a comment for that? Down the line when we come back to these lines things might have changed and it'll help to have a bit of context.

**Review by Anemy — APPROVED** [review]
🔥 🔥 🔥 🔥 🔥 🔥

**Review by lerouxb — APPROVED** [review]
👏

File: packages/node-runtime-worker-thread/package.json [hunk 4]
```diff
@@ -22,8 +22,8 @@
     "test-ci": "node ../../scripts/run-if-package-requested.js npm test",
     "lint": "eslint --report-unused-disable-directives \"./{src,test}/**/*.{js,ts,tsx}\"",
     "check": "npm run lint && depcheck",
-    "webpack-build": "webpack --mode production",
-    "webpack-build-dev": "webpack --mode development",
+    "webpack-build": "npm run compile-ts && webpack --mode production",
+    "webpack-build-dev": "npm run compile-ts && webpack --mode development",
```
[the remark this answers is not in the document]
Comment (gribnoysup) [reply packages/node-runtime-worker-thread/package.json:26]: Huh, based on the `ts-loader` docs it should be an opt-in to generate the d.ts files, so now I'm wondering how it worked before 🤔

[the remark this answers is not in the document]
Comment (addaleax) [reply packages/node-runtime-worker-thread/package.json:26]: Yeah, I don’t know either :woman_shrugging:

File: scripts/docker/rocky8-package.Dockerfile (hunk 3 above)
[the remark this answers is not in the document]
Comment (addaleax) [reply scripts/docker/rocky8-package.Dockerfile:16]: Sure, done! :+1:

**Pull request merged**